pandas/tests/indexing/test_iloc.py

```diff
@@ -706,6 +706,15 @@ def test_iloc_setitem_categorical_updates_inplace(self):
         expected = pd.Categorical(["C", "B", "A"])
         tm.assert_categorical_equal(cat, expected)
 
+        # __setitem__ under the other hand does not work in-place
+        cat = pd.Categorical(["A", "B", "C"])
+        df = pd.DataFrame({1: cat, 2: [1, 2, 3]})
+
+        df[1] = cat[::-1]
+
+        expected = pd.Categorical(["A", "B", "C"])
```

Comment:
can you also check whether the array that ends up in `df[1]` is the same object on the RHS of L713 (Block.setitem is _weird_)